src/CozyUtils.js

```diff
@@ -153,44 +154,101 @@ class CozyUtils {
     let schoolShortcuts = []
     let favShortcuts = []
+
     for (const file of files) {
       if (file.hubMetadata.favori) {
         favShortcuts.push({
           ...file,
-          tempAppId: file.hubMetadata.idInterne,
+          vendorRef: file.hubMetadata.idInterne,
           filename: `${file.title}.url`,
           filestream: `[InternetShortcut]\nURL=${file.url}`,
+          // metadata: {
+          //   icon: svgIcon
+          // },
           shouldReplaceFile: true
         })
       } else {
-        schoolShortcuts.push({
+        const appToSave = {
           ...file,
-          tempAppId: file.hubMetadata.idInterne,
+          vendorRef: file.hubMetadata.idInterne,
           filename: `${file.title}.url`,
           filestream: `[InternetShortcut]\nURL=${file.url}`,
           shouldReplaceFile: true
-        })
+        }
+        if (file.icon) {
+          if (file.icon.match('https://')) {
+            appToSave.metadata = {
+              icon: file.icon
+            }
+          }
+        }
+        schoolShortcuts.push(appToSave)
       }
     }
     return { schoolShortcuts, favShortcuts }
   }
 
-  // Waiting for the full svg icons to be handled
-  async computeThumbnails(files) {
-    let thumbnailsSource = []
-    for (const file of files) {
-      if (file.thumbnail === undefined) {
+  async findShortcuts() {
+    log('info', 'Getting in findShortcuts')
+    // Here we're looking for the shortcuts created by toutatice konnector
+    // We know their path is "/Settings/Home", so there is no need to query the path
+    try {
+      const query = Q('io.cozy.files').partialIndex({
+        type: 'file',
+        class: 'shortcut',
+        'cozyMetadata.createdByApp': 'toutatice',
+        trashed: false
+      })
+      const existingShortcuts = await this.client.queryAll(query)
+      return existingShortcuts
+    } catch (err) {
+      throw new Error(err)
+    }
+  }
+
+  async synchronizeShortcuts(foundShortcuts, computedShortcuts) {
+    let allComputedShortcuts = computedShortcuts.schoolShortcuts.concat(
+      computedShortcuts.favShortcuts
+    )
+    let appsToDelete = []
+    for (let cozyShortcut of foundShortcuts) {
+      if (cozyShortcut.type !== 'file') {
+        log('info', 'is not a file, jumping on next object')
         continue
-      } else if (file.thumbnail.match('.svg')) {
-        thumbnailsSource.push({
-          title: file.title,
-          url: file.thumbnail
-        })
+      }
+      let idx = allComputedShortcuts.findIndex(apiShortcut => {
+        return apiShortcut.vendorRef == cozyShortcut.metadata.fileIdAttributes
+      })
+      if (idx == -1) {
+        appsToDelete.push(cozyShortcut)
+      }
+    }
+    if (appsToDelete.length > 0) {
+      log('info', `${appsToDelete.length} apps to delete`)
+      for (let appToDelete of appsToDelete) {
+        await this.client
+          .collection('io.cozy.files')
+          .deleteFilePermanently(appToDelete._id)
       }
```

Comment:
This code is not very optimal if you have a lot of `appsToDelete` because you map it one by one and make a http call one by one by waiting the result of the call before moving to the other one. 

If you make several calls at the same time, you'll save time execution because call will be send in a more efficient way, but also you'll have the benefit of the http2 mulitplexing. 

So instead, you can write something like: 
```
await Promise.all(appsToDelete.map(appToDelete => this.client
          .collection('io.cozy.files')
          .deleteFilePermanently(appToDelete._id)
))
```

BUT by doing that pretty naïvely, if you have 200 apps to delete, you'll make 200 http requests at the same time. To avoid that, you can use package like `p-limit` that give you the ability to reduce the concurrency.

Reply:
Same here, I was expecting to do that change today, just wanted to push my work on github to save it good.
WIP is added in the title, my bad for forgetting it !